flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaChangelogTableITCase.java

```diff
@@ -369,14 +380,19 @@ public void testKafkaMaxwellChangelogSource() throws Exception {
         Properties producerProperties = getStandardProps();
         producerProperties.setProperty("retries", "0");
         try {
-            stream.addSink(
-                    new FlinkKafkaProducer<>(
-                            topic,
-                            serSchema,
-                            producerProperties,
-                            partitioner,
-                            EXACTLY_ONCE,
-                            DEFAULT_KAFKA_PRODUCERS_POOL_SIZE));
+            stream.sinkTo(
+                    KafkaSink.<String>builder()
+                            .setBootstrapServers(
+                                    producerProperties.getProperty(
+                                            ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))
+                            .setRecordSerializer(
+                                    KafkaRecordSerializationSchema.builder()
+                                            .setTopic(topic)
+                                            .setValueSerializationSchema(serSchema)
+                                            .setPartitioner(partitioner)
+                                            .build())
+                            .setDeliverGuarantee(DeliveryGuarantee.EXACTLY_ONCE)
+                            .build());
```

Comment:
It might be a good idea to extract a private method to avoid triple duplicated code.

Reply:
Unfortunately, the tests are not in great shape I now pulled the sink setup into a separate method but there is still some duplication for the overall testup. I'd leave this for another time ;)